tests/functionality_tests/mission_test.py

```diff
@@ -0,0 +1,54 @@
+from engine.robot import Robot
+from engine.base_station import BaseStation
+from engine.mission import Mission
+from engine.mission import ControlMode
+from engine.database import DataBase
+from engine.robot_state import Robot_State
+from engine.phase import Phase
+
+import math
+
+
+class MissionTest():
+
+    def __init__(self):
+        self.rb_state = Robot_State(0, 0, 0, .2, .5, .2)
+        self.rob = Robot(self.rb_state)
+        self.rob_base = BaseStation(position=(
+            42.444250, -76.483682), heading=math.pi/2, battery=None, plastic_load=None)
+        self.database = DataBase(self.rob)
+        self.m = Mission(robot=self.rob, base_station=self.rob_base,
+                         init_control_mode=ControlMode.LAWNMOWER)
+
+    def test_setup(self):
+        self.rob.execute_setup(self.m.robot_radio_session,
+                               self.m.gps, self.m.imu, self.m.motor_controller)
+        self.rb_state.track_obstacle_thread.stop()
+
+    def test_traverse(self):
```

Comment:
is this being called at all?

Reply:
wdym?

Reply:
is test_traverse being called?

Reply:
the user will change the code in run() to test different mission states. We probably want to test each individually before testing the whole thing at once, which is why I didn't call anything except test_setup in run() rn. Once test_setup() works, user will add the test_traverse() code. Once that is done, user can then add test_docking(), etc. TTD.

Reply:
I'll add a comment noting this test should be run on the physical robot

Reply:
got it! that makes more sense to me (: